beta-src/src/components/wd-modal/WDModal.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from "react";
+import "./WDModal.css";
+import { useRef, useState } from "react";
+import { Popover, Button, Box } from "@mui/material";
+
+interface WDModalProps {
+  ariaDescribedBy: string;
+  children: React.ReactNode;
+  rightPositioning: number; //    Positioning values represent precentages
+  topPositioning: number; //    Positioning values represent precentages
+  triggerIcon: React.ReactNode;
+}
+
+const WDModal: React.FC<WDModalProps> = function ({
+  ariaDescribedBy,
+  children,
+  rightPositioning,
+  topPositioning,
+  triggerIcon,
+}) {
+  const anchorEl = useRef(null);
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleOpen = () => {
+    setIsOpen(true);
+  };
+
+  const handleClose = () => {
+    setIsOpen(false);
+  };
+
+  return (
+    <div
+      className="modal__container"
+      style={{ top: `${topPositioning}%`, right: `${rightPositioning}%` }}
```

Comment:
i think we're going to have a positioning component to handle this, so you don't need this for this PR